### DISCUSSION TITLE
v-pcc tmc2 install

### DISCUSSION THREAD

**Discussion by sCrystalCave** [community/community#60596] [Programming Help]
### Body

https://github.com/MPEGGroup/mpeg-pcc-tmc2

I want to run this code. But there is something I don't know. 
I entered the 4 lines below, but I don't know what to do next. 

mkdir build
cd build
cmake ..
cmake --build . --config Release --parallel 8

the next is External dependencies. 
when I ran the below PccAppEncoder without setting that, this error popped up.

--------------------------------------------------------------------------------------------
yunji@DESKTOP-MAO1J1S:~/mpeg-pcc-tmc2$ ./bin/PccAppEncoder \
iguratio> --configurationFolder=cfg/ \
> --config=cfg/common/ctc-common.cfg \
> --config=cfg/condition/ctc-all-intra.cfg \
fig=cfg/> --config=cfg/sequence/longdress_vox10.cfg \
> --config=cfg/rate/ctc-r3.cfg \
> --uncompressedDataFolder=~/mpeg_datasets/CfP/datasets/Dynamic_Objects/People/ \
> --frameCount=1 \
> --videoEncoderPath=../external/HM-16.16/bin/TAppEncoderStatic \
> --colorSpaceConversionPath=../external/HDRTools/bin/HDRConvert \
> --reconstructedDataPath=S26C03R03_rec_%04d.ply \
> --compressedStreamPath=S26C03R03.bin
PccAppEncoder v18.0

command line error: Unknown option `videoEncoderPath' (value:`../external/HM-16.16/bin/TAppEncoderStatic')
Info: Using external color space conversion
absoluteD1_ should be true when multipleStreams_ is false
absoluteT1 should be true when absoluteD1 is true
------------------------------------------------------------------------------------------

What command should I enter? I'm using Ubuntu 20.04 version.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by hidaviddong** [comment]
same

**Comment by Shriharsh-Deshmukh** [comment]
You're encountering dependency issues. Here's how to resolve it:

1. **Install External Dependencies:**
   - Clone and build the external dependencies (HM-16.16 and HDRTools):
     ```bash
     cd external
     git clone https://vcgit.hhi.fraunhofer.de/jvet/HM.git -b HM-16.16
     git clone https://gitlab.com/standards/HDRTools.git
     cd HM/build
     cmake ..
     make
     cd ../../HDRTools/build
     cmake ..
     make
     ```

2. **Set Correct Paths:**
   - Update the paths in your command to point to the built tools:
     ```bash
     --videoEncoderPath=../external/HM/build/bin/TAppEncoderStatic \
     --colorSpaceConversionPath=../external/HDRTools/build/bin/HDRConvert
     ```

3. **Re-run the Encoder:**
   - After setting the correct paths, re-run your `PccAppEncoder` command.

This should resolve the dependency issue and allow you to run the code.

**Reply by hidaviddong** [reply]
Got it. Many Thanks !

**Reply by hidaviddong** [reply]
> You're encountering dependency issues. Here's how to resolve it:
> 
> 1. **Install External Dependencies:**
>    
>    * Clone and build the external dependencies (HM-16.16 and HDRTools):
>      ```shell
>      cd external
>      git clone https://vcgit.hhi.fraunhofer.de/jvet/HM.git -b HM-16.16
>      git clone https://gitlab.com/standards/HDRTools.git
>      cd HM/build
>      cmake ..
>      make
>      cd ../../HDRTools/build
>      cmake ..
>      make
>      ```
> 2. **Set Correct Paths:**
>    
>    * Update the paths in your command to point to the built tools:
>      ```shell
>      --videoEncoderPath=../external/HM/build/bin/TAppEncoderStatic \
>      --colorSpaceConversionPath=../external/HDRTools/build/bin/HDRConvert
>      ```
> 3. **Re-run the Encoder:**
>    
>    * After setting the correct paths, re-run your `PccAppEncoder` command.
> 
> This should resolve the dependency issue and allow you to run the code.

it doesn't work at branch HM-16.16:
```
cd HM/build
cmake ..
make
```


I ran
```
cd HM/build/linux
make
```
Then generated `TAppEncoderStatic` at `external/HM/bin`

**Reply by Shriharsh-Deshmukh** [reply]
To fix this, adjust your steps as follows:

1. **Switch to the Correct Directory:**
   - Instead of running `cmake` in the `HM/build`, follow the directory structure you mentioned:
     ```bash
     cd external/HM/build/linux
     make
     ```

2. **Update the `videoEncoderPath`:**
   - Since the binary is generated in `external/HM/bin`, adjust your `PccAppEncoder` command:
     ```bash
     --videoEncoderPath=../external/HM/bin/TAppEncoderStatic
     ```

Re-run the command with these changes, and it should work.
